common/src/set-version-labels.js

```diff
@@ -0,0 +1,308 @@
+                    return;
+                }
+
+                response.on('data', (chunk) => {
+                    data += chunk;
+                });
+
+                response.on('end', () => {
+                    resolve(data);
+                });
+            }).on('error', (error) => {
+                reject(error);
+            });
+        });
+    }
+
+    const setLabels = async function (ticketMetadata, labels) {
+        console.log(`Set labels for issue: #${issueNumber}:`, labels);
+        try {
+            await octokit.rest.issues.addLabels({
+                ...ticketMetadata,
+                labels: labels
+            });
+
+            console.log(`Labels set to issue #${issueNumber}:`, labels);
+        } catch (error) {
+            console.error('Error setting labels:', error);
+        }
+    }
+
+    const postGithubComment = async (ticketMetadata, comment) => {
+        await octokit.rest.issues.createComment({
+            ...ticketMetadata,
+            body: comment,
+        });
+    }
+
+    const getCommentText = (potentialToVersionLabelsMap) => {
+        let commentText = "### Set Version Labels Action \n";
+        const potentialLabelsNotMatched = Object.entries(potentialToVersionLabelsMap)
+            .filter(([_, versionLabel]) => versionLabel === null)
+            .map(([potentialLabel, _]) => potentialLabel);
+
+        const patchVersionMissMatch = Object.entries(potentialToVersionLabelsMap)
+            .filter(([potentialLabel, versionLabel]) => {
+                    return versionLabel !== null && potentialLabel.replace('potential:') !== versionLabel.replace('version:')
+                }
+            );
+    
+        if(potentialLabelsNotMatched.length === 0 && patchVersionMissMatch.length === 0) {
+            return null
+        }
+
+        if(potentialLabelsNotMatched.length) {
+            commentText += (
+                "#### Potential Labels with Non-Existing Minor: \n - " +
+                potentialLabelsNotMatched.join("\n - ") + 
+                "\n"
+            );
+        }
+
+        if(patchVersionMissMatch.length) {
+            commentText += (
+                "#### Patch Version Mismatch: \n" +
+                "|Potential Label| Version Label | \n" + 
+                "|---|---| \n" +
+                patchVersionMissMatch
+                    .map(([potentialLabel, versionLabel]) => `|${potentialLabel}|${versionLabel}|`)
+                    .join("\n")
+            );
+        }
+
+        return commentText;
+    }
+
+    const getNoLabelCommentText = () => {
+        return "### Set Version Labels Action \n" + 
+        "Neither valid potential nor valid version label found. Please check if this is intentional.";
+    }
+
+    const hasNoPotentialNorVersionLabels = async (potentialLabels) => {
+        const hasPotentialLabels = (potentialLabels.length > 0);
+
+        if (hasPotentialLabels) {
+            return false;
+        }
```

Comment:
🙃 I think we can skip the constant, right? And since there is a function to check for potential labels, should we use it here?
```suggestion
        if (hasPotentialLabels(potentialLabels)) {
            return false;
        }
```

Reply:
Honoured this point 🙃